### PR TITLE
SE driver: make persistent data work

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Secure element interface changes:

* A driver without a `p_validate_slot_number` method no longer allows registering an existing key with `mbedtls_psa_register_se_key`.
* The `p_validate_slot_number` method now takes an extra read-write parameter `persistent_data`, like `p_allocate`.

New features and bug fixes:
* Call the `p_init` driver method during `psa_crypto_init`.
* Fix loading of the persistent data during init.
* Add some minimal testing (good cases only) for saved persistent data.

### CODE REVIEW

File: include/psa/crypto_se_driver.h [hunk 1]
```diff
@@ -927,7 +927,14 @@ typedef psa_status_t (*psa_drv_se_allocate_key_t)(
  * sake of initial device provisioning or onboarding. Such a mechanism may
  * be added to a future version of the PSA Cryptography API specification.
  *
+ * This function may update the driver's persistent data through
+ * \p persistent_data. The core will save the updated persistent data at the
+ * end of the key creation process. See the description of
+ * ::psa_drv_se_allocate_key_t for more information.
```
Comment (Patater) [inline include/psa/crypto_se_driver.h]: Should we have a test to make sure that the core saved the persistent data?

Comment (gilles-peskine-arm) [reply include/psa/crypto_se_driver.h]: Yes, but that's part of our tech debt. We don't have one for allocate either.

Comment (gilles-peskine-arm) [reply include/psa/crypto_se_driver.h]: I added some testing (good cases only). And fixed a bug that made it not work on deinit/reinit.

**Comment by gilles-peskine-arm** [comment]
CI passed on -head. The failures on -merge are due to a recent CI instability. The PR is from a recent branch and we'd have been happy to merge with a CI job that had run on exactly the same content as the -head job. So CI is ok to merge.

**Pull request merged**